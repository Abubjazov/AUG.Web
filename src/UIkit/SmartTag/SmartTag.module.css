@import url('/src/variables.css');

.root {
  display: flex;
  align-items: center;
  justify-content: center;
  width: fit-content;
  height: 14px;
  padding: 2px 6px;
  font-size: 10px;
  font-weight: 500;
  line-height: 14px;
  color: var(--white);
  text-align: center;
  border-radius: 4px;
}

.label {
  margin-right: 5px;
}

.my-tag {
  background: var(--mytag);
}

.community-tag {
  background: var(--communitytag);
}

.skeleton {
  width: 65px;
  height: 18px;
  padding: 0;
  cursor: not-allowed;
  opacity: 0.5;
  animation: skeleton-loading 1s linear infinite alternate;
}

@keyframes skeleton-loading {
  0% {
    background-color: hsl(198deg 33% 85%);
  }

  100% {
    background-color: hsl(200deg 10% 95%);
  }
}
