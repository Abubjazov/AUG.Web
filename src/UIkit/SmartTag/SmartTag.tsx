import { FC, DragEvent } from 'react'

import { addUserTagToDapplet } from 'store/asyncThunks/userData'
import { useAppDispatch } from 'store/hooks'
import { setModalState } from 'store/slices/layoutSlice'
import Spinner from 'uikit/Spinner/Spinner'
import SvgIcon from 'uikit/SvgIcon'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './SmartTag.module.css'

export enum ESmartTagMode {
  MY_TAG = 'my-tag',
  MY_TAG_MODAL = 'my-tag-modal',
  COMMUNITY_TAG = 'community-tag',
}

export interface SmartTagProps {
  userStyles?: string
  tagId: string
  dappletId?: string
  loading?: boolean
  mode?: ESmartTagMode
  label: string
  onClick?: (tagId: string) => void
}

const SmartTag: FC<SmartTagProps> = ({
  tagId,
  userStyles = '',
  loading = false,
  mode = ESmartTagMode.MY_TAG,
  label,
  onClick,
  dappletId,
}) => {
  const dispatch = useAppDispatch()

  const onDragStartHandler = (event: DragEvent<HTMLDivElement>) => {
    event.stopPropagation()

    event.dataTransfer.setData('tagId', tagId)
    event.dataTransfer.setData('mode', mode)
  }

  const onClickHandler = (event: { stopPropagation: () => void }) => {
    event.stopPropagation()

    if (onClick) onClick(tagId)
  }

  const onDivClickHandler = (event: { stopPropagation: () => void }) => {
    event.stopPropagation()

    if (mode === ESmartTagMode.MY_TAG_MODAL && dappletId) {
      const dragData = {
        dappletId: dappletId,
        userTagId: tagId,
      }

      void dispatch(addUserTagToDapplet(dragData)).finally(() =>
        dispatch(setModalState(false)),
      )
    }
  }

  return (
    <div
      data-testid="smart-tag"
      draggable
      onDragStart={onDragStartHandler}
      className={cc([styles.root, styles[mode], userStyles])}
      onClick={onDivClickHandler}
    >
      {loading ? (
        <div className={styles['spinner-wrapper']}>
          <Spinner width={20} height={20} strokeWidth={8} stroke="#fff" />
        </div>
      ) : (
        <>
          <span>{label}</span>
          {mode !== ESmartTagMode.MY_TAG_MODAL && (
            <button
              data-tag-id={tagId}
              aria-label={`Delete tag ${label}`}
              type="button"
              data-testid={'smart-tag-cross-button-' + tagId}
              className={styles.button}
              onClick={onClickHandler}
            >
              <SvgIcon icon={'smarttagcross'} />
            </button>
          )}
        </>
      )}
    </div>
  )
}

export default SmartTag
