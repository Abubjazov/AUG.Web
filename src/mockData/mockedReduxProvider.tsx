import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import {
  mockCommunityTags,
  mockDapplets,
  mockMyDapplets,
  mockMyTags,
} from './mockData'
import dappletsReducer from '../store/slices/dappletsSlice'
import layoutReducer from '../store/slices/layoutSlice'
import myDappletsReducer from '../store/slices/myDappletsSlice'

export const defaultMockState = {
  layout: {
    menuOpened: false,
    dappletSettingsOpened: false,
  },

  myDapplets: {
    myDapplets: [...mockMyDapplets],
    myTags: [...mockMyTags],
  },

  dapplets: {
    dapplets: [...mockDapplets],
    tags: [...mockCommunityTags],
  },
}

export const mockedReduxProvider = ({
  children,
}: PropsWithChildren<object>) => {
  const store = configureStore({
    reducer: {
      layout: layoutReducer,
      dapplets: dappletsReducer,
      myDapplets: myDappletsReducer,
    },
    preloadedState: {
      ...defaultMockState,
    },
  })

  return <Provider store={store}>{children}</Provider>
}
