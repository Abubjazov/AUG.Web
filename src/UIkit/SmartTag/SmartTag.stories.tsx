import type { Meta, StoryObj } from '@storybook/react'

import SmartTag, { SmartTagProps, SmartTagMode } from './SmartTag'

import '/src/index.css'

const meta: Meta<SmartTagProps> = {
  component: SmartTag,
  title: 'UIkit/Smart Tag',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - draggable smart tag',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
    tagId: { description: 'SmartTag ID' },
    label: {
      description: 'SmartTag contents',
    },
    mode: {
      description: 'SmartTag appearance',
      defaultValue: SmartTagMode.MY_TAG,
    },
    loading: {
      description: 'Appearance of the SmartTag when loading content (skeleton)',
    },
    onClick: {
      description: 'Optional click handler',
    },
  },
}

export default meta

type Story = StoryObj<SmartTagProps>

export const Default: Story = {
  args: {
    tagId: 13,
    label: 'Smart Tag',
  },
}
