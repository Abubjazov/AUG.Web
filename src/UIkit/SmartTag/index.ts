import SmartTag from './SmartTag'

export default SmartTag
