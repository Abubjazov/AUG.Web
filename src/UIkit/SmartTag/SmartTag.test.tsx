import { fireEvent, render, screen } from '@testing-library/react'
import { mockedReduxProvider as Provider } from 'mockData/mockedReduxProvider'
import * as asyncActions from 'services/userData/userData'

import SmartTag, { ESmartTagMode } from './SmartTag'

describe('SmartTag', () => {
  test('should render SmartTag default', () => {
    const { asFragment } = render(
      <Provider>
        <SmartTag tagId={'13'} label={'Smart Tag'} />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render SmartTag "loading"', () => {
    expect(
      render(
        <Provider>
          <SmartTag tagId={'13'} loading label={'Smart Tag'} />
        </Provider>,
      ),
    ).toMatchSnapshot()

    expect(
      render(
        <Provider>
          <SmartTag
            tagId={'13'}
            loading
            mode={ESmartTagMode.MY_TAG}
            label={'Smart Tag'}
          />
        </Provider>,
      ),
    ).toMatchSnapshot()

    expect(
      render(
        <Provider>
          <SmartTag
            tagId={'13'}
            loading
            mode={ESmartTagMode.COMMUNITY_TAG}
            label={'Smart Tag'}
          />
        </Provider>,
      ),
    ).toMatchSnapshot()
  })

  test('should render SmartTag mode: "my tag"', () => {
    const { asFragment } = render(
      <Provider>
        <SmartTag
          tagId={'13'}
          mode={ESmartTagMode.MY_TAG}
          label={'Smart Tag'}
        />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render SmartTag mode: "my tag modal"', () => {
    const { asFragment } = render(
      <Provider>
        <SmartTag
          tagId={'13'}
          mode={ESmartTagMode.MY_TAG_MODAL}
          label={'Smart Tag'}
        />
      </Provider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should call addUserTagToDapplet function on mode: "my tag modal"', () => {
    const mockedAddUserTagToDapplet = vi.spyOn(
      asyncActions,
      'addUserTagToDapplet',
    )

    render(
      <Provider>
        <SmartTag
          dappletId={'d13'}
          tagId={'t13'}
          mode={ESmartTagMode.MY_TAG_MODAL}
          label={'Smart Tag'}
        />
      </Provider>,
    )

    fireEvent.click(screen.getByTestId('smart-tag'))

    expect(mockedAddUserTagToDapplet).toHaveBeenCalledTimes(1)
    expect(mockedAddUserTagToDapplet).toHaveBeenCalledWith({
      dappletId: 'd13',
      userTag: {
        tagId: 't13',
        tagName: 'Smart Tag',
      },
    })
  })

  test('should call onClick function on mode: "my tag"', () => {
    const mockFn = vi.fn()

    render(
      <Provider>
        <SmartTag
          tagId={'13'}
          mode={ESmartTagMode.MY_TAG}
          label={'Smart Tag'}
          onClick={mockFn}
        />
      </Provider>,
    )

    fireEvent.click(screen.getByTestId('smart-tag-cross-button'))

    expect(mockFn).toHaveBeenCalledTimes(1)
  })

  test('should call onDrag function on mode: "my tag"', () => {
    const mockFn = vi.fn()

    render(
      <Provider>
        <SmartTag
          tagId={'13'}
          mode={ESmartTagMode.MY_TAG}
          label={'Smart Tag'}
          onDragStart={mockFn}
        />
      </Provider>,
    )

    fireEvent.dragStart(screen.getByTestId('smart-tag'))

    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})
