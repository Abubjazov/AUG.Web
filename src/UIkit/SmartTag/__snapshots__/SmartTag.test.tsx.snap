// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SmartTag > should render SmartTag "loading" 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_849b90 _my-tag_849b90 "
        data-testid="smart-tag"
        draggable="true"
      >
        <div
          class="_spinner-wrapper_849b90"
        >
          <svg
            class="small-spinner"
            height="20"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 100 100"
            width="20"
          >
            <path
              d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
              fill="none"
              stroke="#fff"
              stroke-dasharray="42.76482137044271 42.76482137044271"
              stroke-linecap="round"
              stroke-width="8"
            >
              <animate
                attributeName="stroke-dashoffset"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0;256.58892822265625"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_849b90 _my-tag_849b90 "
      data-testid="smart-tag"
      draggable="true"
    >
      <div
        class="_spinner-wrapper_849b90"
      >
        <svg
          class="small-spinner"
          height="20"
          preserveAspectRatio="xMidYMid"
          viewBox="0 0 100 100"
          width="20"
        >
          <path
            d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
            fill="none"
            stroke="#fff"
            stroke-dasharray="42.76482137044271 42.76482137044271"
            stroke-linecap="round"
            stroke-width="8"
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="0;256.58892822265625"
            />
          </path>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SmartTag > should render SmartTag "loading" 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_849b90 _my-tag_849b90 "
        data-testid="smart-tag"
        draggable="true"
      >
        <div
          class="_spinner-wrapper_849b90"
        >
          <svg
            class="small-spinner"
            height="20"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 100 100"
            width="20"
          >
            <path
              d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
              fill="none"
              stroke="#fff"
              stroke-dasharray="42.76482137044271 42.76482137044271"
              stroke-linecap="round"
              stroke-width="8"
            >
              <animate
                attributeName="stroke-dashoffset"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0;256.58892822265625"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
    <div>
      <div
        class="_root_849b90 _my-tag_849b90 "
        data-testid="smart-tag"
        draggable="true"
      >
        <div
          class="_spinner-wrapper_849b90"
        >
          <svg
            class="small-spinner"
            height="20"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 100 100"
            width="20"
          >
            <path
              d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
              fill="none"
              stroke="#fff"
              stroke-dasharray="42.76482137044271 42.76482137044271"
              stroke-linecap="round"
              stroke-width="8"
            >
              <animate
                attributeName="stroke-dashoffset"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0;256.58892822265625"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_849b90 _my-tag_849b90 "
      data-testid="smart-tag"
      draggable="true"
    >
      <div
        class="_spinner-wrapper_849b90"
      >
        <svg
          class="small-spinner"
          height="20"
          preserveAspectRatio="xMidYMid"
          viewBox="0 0 100 100"
          width="20"
        >
          <path
            d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
            fill="none"
            stroke="#fff"
            stroke-dasharray="42.76482137044271 42.76482137044271"
            stroke-linecap="round"
            stroke-width="8"
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="0;256.58892822265625"
            />
          </path>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SmartTag > should render SmartTag "loading" 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_849b90 _my-tag_849b90 "
        data-testid="smart-tag"
        draggable="true"
      >
        <div
          class="_spinner-wrapper_849b90"
        >
          <svg
            class="small-spinner"
            height="20"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 100 100"
            width="20"
          >
            <path
              d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
              fill="none"
              stroke="#fff"
              stroke-dasharray="42.76482137044271 42.76482137044271"
              stroke-linecap="round"
              stroke-width="8"
            >
              <animate
                attributeName="stroke-dashoffset"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0;256.58892822265625"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
    <div>
      <div
        class="_root_849b90 _my-tag_849b90 "
        data-testid="smart-tag"
        draggable="true"
      >
        <div
          class="_spinner-wrapper_849b90"
        >
          <svg
            class="small-spinner"
            height="20"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 100 100"
            width="20"
          >
            <path
              d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
              fill="none"
              stroke="#fff"
              stroke-dasharray="42.76482137044271 42.76482137044271"
              stroke-linecap="round"
              stroke-width="8"
            >
              <animate
                attributeName="stroke-dashoffset"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0;256.58892822265625"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
    <div>
      <div
        class="_root_849b90 _community-tag_849b90 "
        data-testid="smart-tag"
        draggable="true"
      >
        <div
          class="_spinner-wrapper_849b90"
        >
          <svg
            class="small-spinner"
            height="20"
            preserveAspectRatio="xMidYMid"
            viewBox="0 0 100 100"
            width="20"
          >
            <path
              d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
              fill="none"
              stroke="#fff"
              stroke-dasharray="42.76482137044271 42.76482137044271"
              stroke-linecap="round"
              stroke-width="8"
            >
              <animate
                attributeName="stroke-dashoffset"
                dur="1s"
                keyTimes="0;1"
                repeatCount="indefinite"
                values="0;256.58892822265625"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_849b90 _community-tag_849b90 "
      data-testid="smart-tag"
      draggable="true"
    >
      <div
        class="_spinner-wrapper_849b90"
      >
        <svg
          class="small-spinner"
          height="20"
          preserveAspectRatio="xMidYMid"
          viewBox="0 0 100 100"
          width="20"
        >
          <path
            d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
            fill="none"
            stroke="#fff"
            stroke-dasharray="42.76482137044271 42.76482137044271"
            stroke-linecap="round"
            stroke-width="8"
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="0;256.58892822265625"
            />
          </path>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SmartTag > should render SmartTag default 1`] = `
<DocumentFragment>
  <div
    class="_root_849b90 _my-tag_849b90 "
    data-testid="smart-tag"
    draggable="true"
  >
    <span>
      Smart Tag
    </span>
    <button
      aria-label="Delete tag Smart Tag"
      class="_button_849b90"
      data-tag-id="13"
      data-testid="smart-tag-cross-button"
      type="button"
    >
      <svg
        fill="none"
        height="7"
        viewBox="0 0 6 7"
        width="6"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_8172_1823)"
        >
          <path
            d="M0.999858 0.887826L2.99986 2.88783L4.99986 0.893886C5.03924 0.853819 5.0865 0.822359 5.13866 0.801495C5.19082 0.780631 5.24674 0.770817 5.30289 0.772674C5.41306 0.779804 5.51688 0.82679 5.59495 0.904859C5.67301 0.982927 5.72 1.08674 5.72713 1.19692C5.72768 1.25109 5.71722 1.30481 5.69638 1.35481C5.67555 1.40482 5.64477 1.45007 5.60592 1.48783L3.59986 3.49995L5.60592 5.51207C5.68471 5.58843 5.72834 5.69389 5.72713 5.80298C5.72 5.91315 5.67301 6.01697 5.59495 6.09504C5.51688 6.1731 5.41306 6.22009 5.30289 6.22722C5.24674 6.22908 5.19082 6.21926 5.13866 6.1984C5.0865 6.17753 5.03924 6.14607 4.99986 6.10601L2.99986 4.11207L1.00592 6.10601C0.966541 6.14607 0.919277 6.17753 0.867117 6.1984C0.814957 6.21926 0.759035 6.22908 0.702888 6.22722C0.590658 6.22138 0.48457 6.17417 0.405104 6.0947C0.325637 6.01523 0.278427 5.90915 0.272585 5.79692C0.272037 5.74275 0.282498 5.68903 0.303334 5.63902C0.32417 5.58902 0.354946 5.54376 0.393797 5.50601L2.39986 3.49995L0.387737 1.48783C0.34998 1.44956 0.320323 1.40408 0.300538 1.3541C0.280753 1.30411 0.271246 1.25066 0.272585 1.19692C0.279715 1.08674 0.326701 0.982927 0.404769 0.904859C0.482838 0.82679 0.586652 0.779804 0.696827 0.772674C0.752538 0.770028 0.808196 0.778887 0.860332 0.798699C0.912468 0.81851 0.959964 0.84885 0.999858 0.887826Z"
            fill="white"
          />
        </g>
        <defs>
          <clippath
            id="clip0_8172_1823"
          >
            <rect
              fill="white"
              height="6"
              transform="translate(0 0.5)"
              width="6"
            />
          </clippath>
        </defs>
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`SmartTag > should render SmartTag mode: "my tag modal" 1`] = `
<DocumentFragment>
  <div
    class="_root_849b90 _my-tag-modal_849b90 "
    data-testid="smart-tag"
    draggable="true"
  >
    <span>
      Smart Tag
    </span>
  </div>
</DocumentFragment>
`;

exports[`SmartTag > should render SmartTag mode: "my tag" 1`] = `
<DocumentFragment>
  <div
    class="_root_849b90 _my-tag_849b90 "
    data-testid="smart-tag"
    draggable="true"
  >
    <span>
      Smart Tag
    </span>
    <button
      aria-label="Delete tag Smart Tag"
      class="_button_849b90"
      data-tag-id="13"
      data-testid="smart-tag-cross-button"
      type="button"
    >
      <svg
        fill="none"
        height="7"
        viewBox="0 0 6 7"
        width="6"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_8172_1823)"
        >
          <path
            d="M0.999858 0.887826L2.99986 2.88783L4.99986 0.893886C5.03924 0.853819 5.0865 0.822359 5.13866 0.801495C5.19082 0.780631 5.24674 0.770817 5.30289 0.772674C5.41306 0.779804 5.51688 0.82679 5.59495 0.904859C5.67301 0.982927 5.72 1.08674 5.72713 1.19692C5.72768 1.25109 5.71722 1.30481 5.69638 1.35481C5.67555 1.40482 5.64477 1.45007 5.60592 1.48783L3.59986 3.49995L5.60592 5.51207C5.68471 5.58843 5.72834 5.69389 5.72713 5.80298C5.72 5.91315 5.67301 6.01697 5.59495 6.09504C5.51688 6.1731 5.41306 6.22009 5.30289 6.22722C5.24674 6.22908 5.19082 6.21926 5.13866 6.1984C5.0865 6.17753 5.03924 6.14607 4.99986 6.10601L2.99986 4.11207L1.00592 6.10601C0.966541 6.14607 0.919277 6.17753 0.867117 6.1984C0.814957 6.21926 0.759035 6.22908 0.702888 6.22722C0.590658 6.22138 0.48457 6.17417 0.405104 6.0947C0.325637 6.01523 0.278427 5.90915 0.272585 5.79692C0.272037 5.74275 0.282498 5.68903 0.303334 5.63902C0.32417 5.58902 0.354946 5.54376 0.393797 5.50601L2.39986 3.49995L0.387737 1.48783C0.34998 1.44956 0.320323 1.40408 0.300538 1.3541C0.280753 1.30411 0.271246 1.25066 0.272585 1.19692C0.279715 1.08674 0.326701 0.982927 0.404769 0.904859C0.482838 0.82679 0.586652 0.779804 0.696827 0.772674C0.752538 0.770028 0.808196 0.778887 0.860332 0.798699C0.912468 0.81851 0.959964 0.84885 0.999858 0.887826Z"
            fill="white"
          />
        </g>
        <defs>
          <clippath
            id="clip0_8172_1823"
          >
            <rect
              fill="white"
              height="6"
              transform="translate(0 0.5)"
              width="6"
            />
          </clippath>
        </defs>
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
